agent/tick_test.go

```diff
@@ -0,0 +1,229 @@
+package agent
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/benbjohnson/clock"
+	"github.com/stretchr/testify/require"
+)
+
+var format = "2006-01-02T15:04:05.999Z07:00"
+
+func TestAlignedTicker(t *testing.T) {
```

Comment:
Do you need to specifically test the system's clock changing? or did I miss that?

Reply:
You didn't miss it.  I tried to do a test for this, but there doesn't appear to be a good way to do it.  Open to any ideas.

Reply:
so basically, you need a time object where the [monotonic clock](https://golang.org/pkg/time/#hdr-Monotonic_Clocks) doesn't change but the wall clock does. I think you'd have to implement the clock.Clock interface, but replace the time.Time with your own time object where you can change the wall clock.